process/smartContract/hooks/vmAccountsDB.go

```diff
@@ -141,26 +184,138 @@ func (vadb *VMAccountsDB) GetCode(address []byte) ([]byte, error) {
 	return code, nil
 }
 
-// GetBlockhash is deprecated
-func (vadb *VMAccountsDB) GetBlockhash(offset *big.Int) ([]byte, error) {
-	return nil, nil
+// GetBlockhash returns the header hash for a requested nonce delta
+func (bh *BlockChainHookImpl) GetBlockhash(offset *big.Int) ([]byte, error) {
+	if offset.Cmp(big.NewInt(0)) > 0 {
+		return nil, process.ErrInvalidNonceRequest
+	}
+
+	hdr := bh.blockChain.GetCurrentBlockHeader()
+
+	requestedNonce := big.NewInt(0).SetUint64(hdr.GetNonce())
+	requestedNonce.Sub(requestedNonce, offset)
+
+	if requestedNonce.Cmp(big.NewInt(0)) < 0 {
+		return nil, process.ErrInvalidNonceRequest
+	}
+
+	if offset.Cmp(big.NewInt(0)) == 0 {
+		return bh.blockChain.GetCurrentBlockHeaderHash(), nil
+	}
+
+	_, hash, err := process.GetShardHeaderFromStorageWithNonce(
```

Comment:
Is it ok? Now you call just the GetShardHeaderWithNonce and not any more GetMetaHeaderWithNonce?